reactive/coredns.py

```diff
@@ -45,5 +88,40 @@ def start_charm():
 
 @when('layer.docker-resource.coredns-image.changed')
 def update_image():
-    # handle a new image resource becoming available
     clear_flag('charm.coredns.started')
+
+
+@when('charm.coredns.started', 'cluster-dns.domain-ready')
+def send_ip():
+    """
+    Send CoreDNS IP to kuberentes-worker
+    """
+    try:
+        cluster_dns = endpoint_from_flag('cluster-dns.domain-ready')
+        if cluster_dns:
+            service_ip = get_service_ip('coredns')
+            if service_ip:
+                cluster_dns.send_ip(service_ip)
+                clear_flag('cluster-dns.domain-ready')
+                layer.status.active('ready')
+            else:
+                layer.status.maintenance('Unable to get service IP')
```

Comment:
This should be a `waiting` status if it's likely to recover on its own, or `blocked` if it will need user intervention. Not `maintenance`.